build/scripts/omega_edit_download.ts

```diff
@@ -15,40 +15,38 @@
  * limitations under the License.
  */
 
-import * as fs from 'fs'
-import * as path from 'path'
-import * as os from 'os'
-import { HttpClient } from 'typed-rest-client/HttpClient'
-import * as crypto from 'crypto'
+// @ts-nocheck <-- This is needed as this file is basically a JavaScript script
+//                 but with some TypeScript niceness baked in
+const fs = require('fs')
+const os = require('os')
+const path = require('path')
+const HttpClient = require('typed-rest-client/HttpClient').HttpClient
+const crypto = require('crypto')
 const hashSum = crypto.createHash('sha512')
 
-// Method to get omega-edit version from a JSON file
-export function getOmegaEditPackageVersion(filePath: fs.PathLike) {
-  return JSON.parse(fs.readFileSync(filePath).toString())['dependencies'][
-    'omega-edit'
-  ]
-}
-
-// Method to get omegaEditServerHash from a JSON file
-export function getOmegaEditServerHash(filePath: fs.PathLike) {
-  return JSON.parse(fs.readFileSync(filePath).toString())['omegaEditServerHash']
-}
+/**
+ * The below classes (Backend, Artifact) are defined similarly in src/classes/artifact.ts.
+ * However for this script to work properly they needed to be redefined in a more JS style.
+ */
 
-export class Backend {
-  constructor(readonly owner: string, readonly repo: string) {}
+class Backend {
+  constructor(owner, repo) {
+    this.owner = owner
+    this.repo = repo
+  }
 }
 
-export class Artifact {
-  constructor(readonly omegaEditVersion: string) {}
-
-  name = `omega-edit-scala-server-${this.omegaEditVersion}`
-  archive = `${this.name}.zip`
-  archiveUrl = (backend: Backend) =>
-    `https://github.com/${backend.owner}/${backend.repo}/releases/download/v${this.omegaEditVersion}/${this.archive}`
-
-  scriptName = os.platform().toLowerCase().startsWith('win32')
-    ? 'example-grpc-server.bat'
-    : './example-grpc-server'
+class Artifact {
+  constructor(omegaEditVersion) {
+    this.omegaEditVersion = omegaEditVersion
+    this.name = `omega-edit-scala-server-${this.omegaEditVersion}`
+    this.archive = `${this.name}.zip`
+    this.archiveUrl = (backend) =>
+      `https://github.com/${backend.owner}/${backend.repo}/releases/download/v${this.omegaEditVersion}/${this.archive}`
```

Comment:
So we can't do that because the `omega-edit` scala server is only packaged using `universal:packageBin`, like we do with the bugger, so creates a zip with the needed scripts and JAR files. Moreover, this ZIP not published to any maven/sbt repos it is only published to the GitHub release page. So this file is not building anything necessarily it is only downloading the `omega-edit` scala server from the github releases.

Reply:
I took a look and I didn't find exactly the thing I wanted which is sbt dependencies that look for github releases. You can do source code dependencies, but not binary dependencies.